### ISSUE TITLE
Keys and values are out of sync in coreconsole.cpp

### ISSUE THREAD

**Issue by moozzyk** [dotnet/coreclr#1676]
When initializing the runtime in coreconsole the number of keys passed is 5 - https://github.com/dotnet/coreclr/blob/8ff2389204d7c41b17eff0e9536267aea8d6496f/src/coreclr/hosts/coreconsole/coreconsole.cpp#L455-L460 
but the number of values is 4
https://github.com/dotnet/coreclr/blob/8ff2389204d7c41b17eff0e9536267aea8d6496f/src/coreclr/hosts/coreconsole/coreconsole.cpp#L462-L471
Since the number of properties is calculated based on the size of the array that contains keys (https://github.com/dotnet/coreclr/blob/8ff2389204d7c41b17eff0e9536267aea8d6496f/src/coreclr/hosts/coreconsole/coreconsole.cpp#L503) trying to read the value of the "AppDomainCompatSwitch" key may segfault.

Disclaimer - I don't exactly know what this code does and have not tried running it. I found this by accident and this just does not seem right.

**Comment by janvorli** [comment]
It is actually broken in yet another way. Looking at the keys, there is a missing comma after the `W("NATIVE_DLL_SEARCH_DIRECTORIES")`, so the actual key is NATIVE_DLL_SEARCH_DIRECTORIESAppDomainCompatSwitch and that makes keys and values the same size.

```C
    const wchar_t *property_keys[] = { 
        W("TRUSTED_PLATFORM_ASSEMBLIES"),
        W("APP_PATHS"),
        W("APP_NI_PATHS"),
        W("NATIVE_DLL_SEARCH_DIRECTORIES")
		W("AppDomainCompatSwitch"),
    };
```

**Comment by janvorli** [comment]
@jkotas, this is the windows version of coreconsole. Do we want to pass in any value of the AppDomainCompatSwitch or should we just remove it from the propery list?

**Comment by janvorli** [comment]
@moozzyk Thanks spotting and reporting it!

**Comment by jkotas** [comment]
The value for AppDomainCompatSwitch should be UseLatestBehaviorWhenTFMNotSpecified - it is what we do everywhere, except this one place.

**Comment by moozzyk** [comment]
LOL - indeed both arrays have the same number of items because of the missing coma. This is probably why things did not go awry completely.

**Comment by janvorli** [comment]
Fixed by #1726
